src/Service.Tests/OpenApiDocumentor/ParameterValidationTests.cs

```diff
@@ -0,0 +1,187 @@
+
+    /// <summary>
+    /// Validates the path parameters for GET methods.
+    /// </summary>
+    /// <param name="entityName">The name of the entity.</param>
+    /// <param name="objectName">The name of the database object.</param>
+    /// <param name="entitySourceType">The source type of the entity.</param>
+    [DataTestMethod]
+    [DataRow("BooksTable", "books", EntitySourceType.Table, DisplayName = "Table with path parameter id")]
+    [DataRow("BooksView", "books_view_all", EntitySourceType.View, DisplayName = "View with path parameter id")]
+    public async Task ValidatePathParametersForTablesAndViews(string entityName, string objectName, EntitySourceType entitySourceType)
+    {
+        EntitySource entitySource = new(Object: objectName, entitySourceType, null, null);
+        OpenApiDocument openApiDocument = await GenerateOpenApiDocumentForGivenEntityAsync(entityName, entitySource);
+
+        foreach ((string pathName, OpenApiPathItem pathItem) in openApiDocument.Paths)
+        {
+            string apiPathWithParam = $"/{entityName}/id/{{id}}";
+            if (pathName.Equals(apiPathWithParam))
+            {
+                Assert.IsTrue(pathItem.Parameters.Count is 1);
+                OpenApiParameter pathParameter = pathItem.Parameters.First();
+                Assert.AreEqual("id", pathParameter.Name);
+                Assert.AreEqual(ParameterLocation.Path, pathParameter.In);
+                Assert.IsTrue(pathParameter.Required);
+            }
+            else
+            {
+                // Get All and POST method with path /entityName, will have no path parameters.
+                Assert.IsTrue(pathItem.Parameters.Count is 0);
+            }
+        }
+    }
+
+    /// <summary>
+    /// Validates that Query parameters are generated for GET methods in Table/Views.
+    /// $select, $filter, $orderby, $first, $after are the query parameters.
+    /// </summary>
+    /// <param name="entityName">The name of the entity.</param>
+    /// <param name="objectName">The name of the database object.</param>
+    /// <param name="entitySourceType">The source type of the entity.</param>
+    [DataTestMethod]
+    [DataRow("BooksTable", "books", EntitySourceType.Table, DisplayName = "Table with query parameters")]
+    [DataRow("BooksView", "books_view_all", EntitySourceType.View, DisplayName = "View with query parameters")]
+    public async Task ValidateQueryParametersForTablesAndViews(string entityName, string objectName, EntitySourceType entitySourceType)
+    {
+        EntitySource entitySource = new(Object: objectName, entitySourceType, null, null);
+        OpenApiDocument openApiDocument = await GenerateOpenApiDocumentForGivenEntityAsync(entityName, entitySource);
+
+        foreach (OpenApiPathItem pathItem in openApiDocument.Paths.Values)
+        {
+            foreach ((OperationType operationType, OpenApiOperation operation) in pathItem.Operations)
```

Comment:
instead of having loops, especially a double for loop, could you try to 'get' the expected path item from `openApiDocument.Paths.Values` and then 
validate you retrieve an object AND the expected number of objects.

Then, you can 'get' the expected tuple of operationType and operation to
- validate an expected number of tuples exists in `pathItem.Operations`
- perform your pre-existing assertions.
- assert that GET by ID does not contain query parameters irrelevant to single object query. (e.g. filter, orderby, first, after aren't applicable to get by id). 

For easier reading, I would argue that you have a separate test which validates zero query parameters exist for non-GET operation types.
- Test 1 -> `QueryParametersAddedForGEToperationOnTablesAndViews`
- Test 2 -> `QueryParametersExcludedFromNonReadOperationsOnTablesAndViews` -> this test would then explicitly validate lack of query params for POST, PUT, PATCH, DELETE, and GET (by ID)

Reply:
```
Test 2 -> QueryParametersExcludedFromNonReadOperationsOnTablesAndViews -> this test would then explicitly validate lack of query params for POST, PUT, PATCH, DELETE, and GET (by ID)
```
why GET (by ID) should not have query params? I agree some parameters won't makes sense with it, but parameters like $select does. I can see if there is a way to add a condition.

Reply:
When you're retrieving a single resource by its ID, these parameters generally don't make sense because there's only one resource to retrieve, so there's no need to filter or sort the results, and it's usually desirable to return all of the resource's fields.

However, there might be some cases where $select could be useful in a getById request.

Reply:
Yeah, agreed. I should have been more explicit, but what i meant with "GET by ID does not contain query parameters irrelevant to single object query" was that only query params that make sense like `$select` that you mention, should be applicable.